pkg/ebpf/bpf_client.go

```diff
@@ -541,10 +540,12 @@ func (l *bpfClient) DetacheBPFProbes(pod types.NamespacedName, ingress bool, egr
 				sdkAPIErr.WithLabelValues("deleteBPFProgramAndMaps").Inc()
 			}
 			l.policyEndpointeBPFContext.Delete(podIdentifier)
+			if _, ok := l.AttachProbesToPodLock.Load(podIdentifier); ok {
+				l.AttachProbesToPodLock.Delete(podIdentifier)
+			}
 		}
 		l.DeletePodFromEgressProgPodCaches(pod.Name, pod.Namespace)
 	}
-	l.DeletePodFromAttachProbesToPodLock(pod.Name, pod.Namespace)
```

Comment:
Why do we not need this now? (Because the Ingress and Egress Progs will be consistent?)

Reply:
@orsenthil Earlier as we are storing the lock per pod, we were deleting it whenever we remove probes for the pod (pod no longer exists or no policies applied to the pod). Now because the lock is per pod identifier, we should delete it if this is the last pod in the pod identifier. 
So moved the delete to here 
```
if deletePinPath {
----->
			l.policyEndpointeBPFContext.Delete(podIdentifier)
			if _, ok := l.AttachProbesToPodLock.Load(podIdentifier); ok {
				l.AttachProbesToPodLock.Delete(podIdentifier)
			}
----->
```

deletePinPath will be set to true when this is the last pod in the podidentifier. We delete pinpath, progFDs and other ds related to podIdentifier when this condition is true

Reply:
Okay, this is only "lock" delete. I got confused a bit. 

Thanks for the explanation. LGTM.